torchgen/native_function_generation.py

```diff
@@ -103,27 +105,32 @@ def self_to_out_signature(func: FunctionSchema) -> FunctionSchema:
     )
 
 
-# Helper function: given a mutable FunctionSchema, generate its corresponding out= variant
+# Helper function: given a functional FunctionSchema, generate its corresponding out= variant
 # Example before:
-#   _fused_moving_avg_obs_fq_helper(Tensor self, Tensor observer_on, Tensor fake_quant_on, Tensor(a!) running_min, Tensor(b!) running_max, Tensor(c!) scale, Tensor(d!) zero_point, float averaging_const, int quant_min, int quant_max, int ch_axis, bool per_row_fake_quant=False, bool symmetric_quant=False) -> (Tensor output, Tensor mask)  # noqa: B950
+#   _to_copy(Tensor self, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None, bool non_blocking=False, MemoryFormat? memory_format=None) -> Tensor
```

Comment:
ah ok - I think we should it looks like `_to_copy()` takes in the full set of TensorOptions arguments.

I think that to fit the pattern that we have for other factory functions with out= variants, we should arrange for the signature to look like:
```
_to_copy.out(Tensor self, *, bool non_blocking=False, MemoryFormat? memory_format=None, Tensor(a!) out) -> Tensor(a!)
```

Reply:
ok, if we set `_to_copy.out` signature like that, `autogen` won't work on it?